Task/create meeting router

Matios102 opened the pull request.

Matios102 committed:
Initiali commit

Matios102 committed:
Create meeting: router, service, repository

mateuszpolis commented on .flake8:
```diff
@@ -0,0 +1,5 @@
+[flake8]
```
if you installed something do pip freeze > requirements.txt

mateuszpolis commented on Dockerfile:
```diff
@@ -1,7 +1,8 @@
 FROM python:3.9-slim
 WORKDIR /app
-COPY . /app
```
why did you move that around

HermanPlay replied:
It was me, when after this change you don't have to install requirements every time you build, the build will only copy new files and build the image and use cached requirements

mateuszpolis commented on models/__init__.py:
```diff
@@ -0,0 +1,6 @@
+from .attendance import attendance
+from .classes import classes
```
put models back in one file as it was

mateuszpolis commented on models/meetings.py:
```diff
@@ -0,0 +1,21 @@
+    Column("id", Integer, primary_key=True, index=True),
+    Column("teacher_id", Integer, ForeignKey("users.id")),
+    Column("class_id", Integer, ForeignKey("classes.id")),
+    Column("start_time", String),
```
remove start_time and end_time from here, its already in class

mateuszpolis commented on repositories/meeting_repository.py:
```diff
@@ -0,0 +1,36 @@
+def get_meeting_by_current_time(db: Session, current_time: datetime):
+    meeting_query = select(meetings).where(
+        and_(
+            meetings.c.start_time <= current_time.strftime("%H:%M"),
```
get start_time and end_time from class

mateuszpolis commented on repositories/meeting_repository.py:
```diff
@@ -0,0 +1,36 @@
+    )
+    return db.execute(query).fetchall()
+
+def cancel_meeting_by_id(db: Session, meeting_id: int):
```
cancel does not mean delete. Add column canceled of type bool to the table and set that to true, then delete attendance for that meeting if there were any

mateuszpolis commented on routers/meetings_router.py:
```diff
@@ -0,0 +1,40 @@
+):
+    return fetch_meeting_by_id(db, meeting_id, current_user.id)
+
+@router.get("/meetings/{meeting_id}/students", response_model=List[StudentBase])
```
We don't want StudentBase here, we want to know if the student was present or not. Create a new schema that extends StudentBase with added field preset

mateuszpolis commented on routers/meetings_router.py:
```diff
@@ -0,0 +1,40 @@
+):
+    return fetch_meeting_by_current_time(db, current_user.id)
+
+@router.get("/meetings/{meeting_id}")
```
combine this meetings/{meeting_id} with meetings/{meeting_id}/students.
When we request the meeting we already want the list of students. 

You can look in classes how to join them

Matios102 committed:
fixed pointed out issues except the last two

Matios102 committed:
fixes

Matios102 committed:
feat: Add meeting details with students to meeting_repository

This commit adds a new function `get_meeting_by_id_with_students` to the `meeting_repository.py` file. This function retrieves a meeting by its ID and includes the details of the students attending the meeting. It also fetches the attendance data for each student.

The function uses SQLAlchemy to construct the necessary SQL queries to retrieve the meeting and student data. It joins the `students` and `students_classes` tables to get the students attending the meeting, and the `attendance` table to get the attendance data for each student.

This new function will be useful for displaying the meeting details along with the students and their attendance information in the application.

Note: The commit message has been generated based on the provided code changes and recent commits.
